### PR TITLE
Own solution

### PR DESCRIPTION
**Pull request by wielkiborsuk**

### CODE REVIEW

File: src/screens/Movies.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    title: 'Movie list'
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      movies: []
+    }
+
+    movieService.getMovies()
```
Comment (kamilszkutnik) [inline src/screens/Movies.js:17]: I think that the best place to fetch data is ComponentDidMount. There are two reasons: - "Using DidMount makes it clear that data won’t be loaded until after the initial render. This reminds you to set up initial state properly, so you don’t end up with undefined state that causes errors." and "If you ever need to render your app on the server (SSR/isomorphic/other buzzwords), componentWillMount will actually be called twice – once on the server, and again on the client – which is probably not what you want. Putting the data loading code in componentDidMount will ensure that data is only fetched from the client." https://daveceddia.com/where-fetch-data-componentwillmount-vs-componentdidmount/

File: src/services/MovieService.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class MovieService {
+  getMovies() {
+    return fetch('http://api.tvmaze.com/shows')
+      .then((response) => response.json());
+  }
+}
+
+export { MovieService };
+export default new MovieService();
```
Comment (kamilszkutnik) [inline src/services/MovieService.js:9]: Do you need those two exports here?

File: src/screens/Movies.js (hunk 1 above)
Comment (wielkiborsuk) [reply src/screens/Movies.js:17]: true, sure - will remember, initializing in the constructor is an old non-react habit

File: src/services/MovieService.js (hunk 2 above)
Comment (wielkiborsuk) [reply src/services/MovieService.js:9]: Not really, at first i exported the class, then switched to an instance, but never cleaned up the class export

**Pull request closed without merging**